test/hotspot/jtreg/compiler/lib/template_framework/Template.java

```diff
@@ -0,0 +1,692 @@
+ * to be used in code, as well as its type and if it is mutable. One can add such a {@link Name} to the
+ * current code scope with {@link #addName}, and sample from the current or outer scopes with {@link #sampleName}.
+ * When generating code, one might want to create {@link Name}s (variables, fields, etc.) in local scope, or
+ * in some outer scope with the use of {@link Hook}s.
+ *
+ * <p>
+ * More examples for these functionalities can be found in {@link TestTutorial}, {@link TestSimple}, and
+ * {@link TestAdvanced}.
+ */
+public sealed interface Template permits Template.ZeroArgs,
+                                         Template.OneArgs,
+                                         Template.TwoArgs,
+                                         Template.ThreeArgs {
+
+    /**
+     * A {@link Template} with no arguments.
+     *
+     * @param function The {@link Supplier} that creates the {@link TemplateBody}.
+     */
+    public record ZeroArgs(Supplier<TemplateBody> function) implements Template {
+        TemplateBody instantiate() {
+            return function.get();
+        }
+
+        /**
+         * Creates a {@link TemplateToken} which can be used as a {@link Token} inside
+         * a {@link Template} for nested code generation.
+         *
+         * @return The template all (zero) arguments applied.
+         */
+        public TemplateToken asToken() {
+            return new TemplateToken.ZeroArgs(this);
+        }
+
+        /**
+         * Renders the {@link Template} to String.
+         *
+         * @return The String, resulting from rendering the {@link Template}.
+         */
+        public String render() {
+            return new TemplateToken.ZeroArgs(this).render();
+        }
+
+        /**
+         * Renders the {@link Template} to String.
+         *
+         * @param fuel The amount of fuel provided for recursive Template instantiations.
+         * @return The String, resulting from rendering the {@link Template}.
+         */
+        public String render(float fuel) {
+            return new TemplateToken.ZeroArgs(this).render(fuel);
+        }
+    }
+
+    /**
+     * A {@link Template} with one argument.
+     *
+     * @param arg0Name The name of the (first) argument, used for hashtag replacements in the {@link Template}.
+     * @param <A> The type of the (first) argument.
+     * @param function The {@link Function} that creates the {@link TemplateBody} given the template argument.
+     */
+    public record OneArgs<A>(String arg0Name, Function<A, TemplateBody> function) implements Template {
+        TemplateBody instantiate(A a) {
+            return function.apply(a);
+        }
+
+        /**
+         * Creates a {@link TemplateToken} which can be used as a {@link Token} inside
+         * a {@link Template} for nested code generation.
+         *
+         * @param a The value for the (first) argument.
+         * @return The template its argument applied.
```

Comment:
```suggestion
         * @return The template with its argument applied.
```

Reply:
Changed the wording completely, because we now return a `TemplateToken` :)